storage_controller/src/service.rs

```diff
@@ -6403,18 +6407,39 @@ impl Service {
         // TODO: issue split calls concurrently (this only matters once we're splitting
         // N>1 shards into M shards -- initially we're usually splitting 1 shard into N).
 
+        // HADRON: set a timeout for splitting individual shards on page servers.
+        // Currently we do not perform any retry because it's not clear if page server can handle
+        // partially split shards correctly.
+        let shard_split_timeout =
```

Comment:
TODO: This seems like a quick hack to get it working. I'd suggest having a single place to generate all such HTTP clients (maybe @DimasKovas can take a look in the future) and also reuse HTTP clients to avoid exhausting ports; fine to merge as-is, no need to block

Reply:
You can just use `tokio::timeout` over `client.tenant_shard_split()` future

But yeah, it's fine to merge as is